h2o-core/src/main/java/hex/ModelMetricsBinomialUplift.java

```diff
@@ -0,0 +1,294 @@
+                    resp = m==null && frameWithExtraColumns.vec(f.numCols()-1).isCategorical() ?
+                            frameWithExtraColumns.vec(f.numCols()-1) //work-around for the case where we don't have a model, assume that the last column is the actual response
+                            :
+                            frameWithExtraColumns.vec(m._parms._response_column);
+                    if (resp != null) {
+                        weight = m==null?null : frameWithExtraColumns.vec(m._parms._weights_column);
+                    }
+                    if(m != null && m._parms._treatment_column != null){
+                        treatment = frameWithExtraColumns.vec(m._parms._treatment_column);
+                    }
+                }
+            }
+            return makeModelMetrics(m, f, preds, resp, weight, treatment);
+        }
+
+        private ModelMetrics makeModelMetrics(final Model m, final Frame f, final Frame preds,
+                                              final Vec resp, final Vec weight, final Vec treatment) {
+            GainsUplift gul = null;
+            AUUC auuc = null;
+            if (_wcount > 0) {
+                if (preds != null) {
+                    if (resp != null) {
+                        if (_auuc == null) {
+                            auuc = new AUUC(preds.vec(0), resp, treatment, m._parms._auuc_type);
+                        }
+                        final Optional<GainsUplift> optionalGainsUplift = calculateGainsUplift(m, preds, resp, weight, treatment);
+                        if (optionalGainsUplift.isPresent()) {
+                            gul = optionalGainsUplift.get();
+                        }
+                    }
+                }
+            }
+            return makeModelMetrics(m, f, auuc, gul);
+        }
+
+        private ModelMetrics makeModelMetrics(Model m, Frame f, AUUC auuc, GainsUplift gul) {
+            double sigma = Double.NaN;
+            if(_wcount > 0 && auuc == null) {
+                sigma = weightedSigma();
+                auuc = new AUUC(_auuc, m._parms._auuc_type);
+            } else if(auuc == null){
+                auuc = new AUUC();
+            }
+            ModelMetricsBinomialUplift mm = new ModelMetricsBinomialUplift(m, f, _count, _domain, sigma, auuc, gul, _customMetric);
+            if (m!=null) m.addModelMetrics(mm);
+            return mm;
+        }
+
+        /**
+         * @param m       Model to calculate GainsUplift for
+         * @param preds   Predictions
+         * @param resp    Actual label
+         * @param weights Weights
+         * @param treatment  Treatment column               
+         * @return An Optional with GainsUplift instance if GainsUplift is not disabled (gainsUplift_bins = 0). Otherwise an
+         * empty Optional.
+         */
+        private Optional<GainsUplift> calculateGainsUplift(Model m, Frame preds, Vec resp, Vec weights, Vec treatment) {
```

Comment:
since only `isPresent` method is called on the result, using `Optional` doesn't make sense and is equivalent to returning `null` or the result directly and performing `null` comparison without an additional wrapper